### PR TITLE
Correct assignment to SetHash

### PR DESCRIPTION
**Pull request by vrurg**
rakudo/rakudo#1203

### CODE REVIEW

File: src/core/SetHash.pm6 [hunk 1]
```diff
@@ -203,6 +203,14 @@ my class SetHash does Setty {
     multi method Mixy (SetHash:D:) { self.MixHash }
 
 #--- interface methods
+    multi method STORE(SetHash:D: Setty:D \set --> SetHash:D) {
+        nqp::say("SetHash STORE from Setty") if nqp::getenvhash<RAKUDO_DEBUG>;
```
Comment (MasterDuke17) [inline src/core/SetHash.pm6]: This looks like a debugging leftover.

Comment (vrurg) [reply src/core/SetHash.pm6]: It surely does. Thanks!

**Comment by MasterDuke17** [comment]
It also looks like this is pulling in some commits related to #2178?

File: src/core/SetHash.pm6 [hunk 2]
```diff
@@ -203,6 +203,14 @@ my class SetHash does Setty {
...
+        self.SET-SELF(
```
Comment (MasterDuke17) [inline src/core/SetHash.pm6]: Also, this new candidate doesn't need the `.is-lazy` check that the existing candidate has?

Comment (vrurg) [reply src/core/SetHash.pm6]: I think a `Setty` cannot be lazy.

**Comment by vrurg** [comment]
Looks like I need to redo this PR and rebase it on master. Thank for noticing!

**Pull request merged**